### PR TITLE
1-D array in Python added

### LINKED ISSUE #1320: Add 1-D Array in python
1-D Array in C and C++ is present. I want to add it in python.

### PR DESCRIPTION
**Pull request by divya1999ann**
Fixes #1320 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- 1-d array in python added

#### Languages Used:
- python

#### Files Added:
- Frequency_Of_A_Element.py
- Max_And_Next_Max_Number.py
- Mean_Variance_Standard_Deviation.py
- Merge_Two_Sorted_Arrays.py

### CODE REVIEW

File: 1-D_Array/Python/Frequency_Of_A_Element.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+num = int(input("Enter number of elements in the array"))
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Frequency_Of_A_Element.py]: @divya1999ann there are 4 files in this PR. Can you please remove the files not related to this PR before we review this?

**Comment by divya1999ann** [comment]
I had added the folder of 1-D Array in python which has all these 4 files.

File: 1-D_Array/Python/Frequency_Of_A_Element.py (hunk 1 above)
Comment (marwahmanbir) [inline 1-D_Array/Python/Frequency_Of_A_Element.py]: Add a comment at the top to describe what this program does.

File: 1-D_Array/Python/Frequency_Of_A_Element.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+1
+1
+2
+2
+3
+Enter the number whose occurence is to be found: 3
+Frequency is
+1
+'''
+
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Frequency_Of_A_Element.py]: Remove extra line, there are two lines at the end of file, keep only one.

File: 1-D_Array/Python/Max_And_Next_Max_Number.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# Comparing rest of the element in the array
+for i in range(2, num):
+    if arr[i] > smax:
+        if arr[i] > max:
+            smax = max				
+            max = arr[i]
+        else:
+            smax = arr[i]
+
+
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Max_And_Next_Max_Number.py]: Remove an extra line.

File: 1-D_Array/Python/Max_And_Next_Max_Number.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+Enter your array:
+1
+5
+6
+3
+4
+Max:  6
+Next max:  5
+'''
+
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Max_And_Next_Max_Number.py]: Remove extra line.

File: 1-D_Array/Python/Mean_Variance_Standard_Deviation.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+#Python program to find mean , variance and standard deviation
+
+num = int(input("Enter number of elements in array: "))
+arr=[]
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Mean_Variance_Standard_Deviation.py]: Space around ``=``.

File: 1-D_Array/Python/Mean_Variance_Standard_Deviation.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+print("Enter your array: ")
+for i in range(0, num):
+    arr.insert(i, float(input()))
+
+
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Mean_Variance_Standard_Deviation.py]: Extra

File: 1-D_Array/Python/Mean_Variance_Standard_Deviation.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+# mean is the average of the numbers
+mean = mean / num
+
+sum = 0.0;
+
+for i in range(0, num):
+    sum = sum + (arr[i] - mean) ** 2
+
+
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Mean_Variance_Standard_Deviation.py]: Extra

File: 1-D_Array/Python/Max_And_Next_Max_Number.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
+#Python program to find the maximum and the next maximum number in an array
+
+num = int(input("Enter number of elements in array: "))
+arr = []
+print("Enter your array: ");
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Max_And_Next_Max_Number.py]: Semicolons are not required in Python to terminate a statement as far as I know. You are using them in all files.

File: 1-D_Array/Python/Mean_Variance_Standard_Deviation.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+Enter your array:
+1
+2
+3
+4
+5
+Mean =  3.0
+Variance =  2.0
+Standard Deviation =  1.4142135623730951
+'''
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Mean_Variance_Standard_Deviation.py]: Add a single line at the end of file.

File: 1-D_Array/Python/Merge_Two_Sorted_Arrays.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
+#Python program to merge two sorted array a[] and b[] and storing it in c[]
+
+a=[]
+b=[]
+c=[]
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Merge_Two_Sorted_Arrays.py]: Spaces around ``=`` in above 3 statements.

File: 1-D_Array/Python/Merge_Two_Sorted_Arrays.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    c[k++] = a[i++]
+
+# copying remaining elements of array b
+while j < num2 :
+    c[k++] = b[j++]
+
+print("New merged array: ");
+for i in c:
+    print(i)
+
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Merge_Two_Sorted_Arrays.py]: Add sample input-output.

File: 1-D_Array/Python/Merge_Two_Sorted_Arrays.py [hunk 12]
```diff
@@ -0,0 +1,37 @@
+#Python program to merge two sorted array a[] and b[] and storing it in c[]
```
Comment (marwahmanbir) [inline 1-D_Array/Python/Merge_Two_Sorted_Arrays.py]: No import statements required in any file?

Comment (divya1999ann) [reply 1-D_Array/Python/Merge_Two_Sorted_Arrays.py]: Nope

**Comment by divya1999ann** [comment]
I have made the changes , please check .

File: 1-D_Array/Python/Frequency_Of_A_Element.py [hunk 13]
```diff
@@ -0,0 +1,28 @@
+# Python program to find the frequency of an element
+
+num = int(input("Enter number of elements in the array"))
+arr = []
+f = 0
+for i in range(0, num):
+    arr.insert(i, int(input()))
+
+#Finding frequency of an element
+n = int(input ("Enter the number whose occurence is to be found: "))
```
Comment (MastersAbh) [inline 1-D_Array/Python/Frequency_Of_A_Element.py]: please remove space before (

File: 1-D_Array/Python/Frequency_Of_A_Element.py [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+for i in arr:
+    if i == n :
```
Comment (MastersAbh) [inline 1-D_Array/Python/Frequency_Of_A_Element.py]: no space before : in python

File: 1-D_Array/Python/Mean_Variance_Standard_Deviation.py [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+
+sum = 0.0;
+
+for i in range(0, num):
+    sum = sum + (arr[i] - mean) ** 2
+
+# Variance is the average of the squared differences from the Mean
+v = sum / num
+# Standard Deviation is a measure of how spread out numbers are, it is the square root of the Variance.
+sd = v ** (1/2);
```
Comment (MastersAbh) [inline 1-D_Array/Python/Mean_Variance_Standard_Deviation.py]: space around /

File: 1-D_Array/Python/Merge_Two_Sorted_Arrays.py [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+for i in range(0, num2):
+    b.insert(i, int(input())
+
+k = 0;
+j = 0;
+i = 0;
+
+# Traverse both array 
+while i < num1 AND j < num2 :
+    if a[i] > b[j] :
```
Comment (MastersAbh) [inline 1-D_Array/Python/Merge_Two_Sorted_Arrays.py]: no space before :

**Comment by divya1999ann** [comment]
made changes

File: 1-D_Array/Python/Mean_Variance_Standard_Deviation.py [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+mean = 0.0
+
+for i in range(0, num):
+    # Calculating sum of the array
+    mean = mean + arr[i]
+
+# mean is the average of the numbers
+mean = mean / num
+
+sum = 0.0;
```
Comment (MastersAbh) [inline 1-D_Array/Python/Mean_Variance_Standard_Deviation.py]: no semicolon

File: 1-D_Array/Python/Mean_Variance_Standard_Deviation.py [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+
+for i in range(0, num):
+    sum = sum + (arr[i] - mean) ** 2
+
+# Variance is the average of the squared differences from the Mean
+v = sum / num
+# Standard Deviation is a measure of how spread out numbers are, it is the square root of the Variance.
+sd = v ** (1 / 2);
```
Comment (MastersAbh) [inline 1-D_Array/Python/Mean_Variance_Standard_Deviation.py]: no semicolon

**Comment by divya1999ann** [comment]
done

**Comment by MastersAbh** [comment]
Great work. please squash your commits.

**Comment by divya1999ann** [comment]
yes done

**Pull request merged**